python/tests/test_lowlevel.py

```diff
@@ -2673,6 +2673,102 @@ def test_options(self):
             )
 
 
+class TestGeneticRelatednessVector(LowLevelTestCase):
+    def get_example(self, num_weights=2):
+        ts = self.get_example_tree_sequence()
+        num_samples = ts.get_num_samples()
+        params = {
+            "weights": np.linspace(0, 1, num_weights * num_samples).reshape(
+                (num_samples, num_weights)
+            ),
+            "windows": [0, ts.get_sequence_length()],
+        }
+        return ts, params
+
+    @pytest.mark.parametrize(
+        "mode",
+        [
+            "branch",
+        ],
+    )
+    @pytest.mark.parametrize("num_weights", [1, 3])
+    def test_basic_example(self, mode, num_weights):
+        ts, params = self.get_example(num_weights)
+        ns = ts.get_num_samples()
+        result = ts.genetic_relatedness_vector(
+            params["weights"], params["windows"], mode, True, True
+        )
+        assert result.shape == (1, ns, num_weights)
+        result = ts.genetic_relatedness_vector(
+            params["weights"], params["windows"], mode, True, False
+        )
+        assert result.shape == (1, ns, num_weights)
+        result = ts.genetic_relatedness_vector(
+            params["weights"], params["windows"], mode, False, True
+        )
+        assert result.shape == (1, ns, num_weights)
+
+    @pytest.mark.parametrize("mode", ["site", "node"])
+    def test_modes_not_supported(self, mode):
+        ts, params = self.get_example()
+        with pytest.raises(_tskit.LibraryError):
+            ts.genetic_relatedness_vector(
+                params["weights"], params["windows"], mode, True, True
+            )
+
+    @pytest.mark.parametrize(
+        "mode",
+        [
+            "branch",
+        ],
```

Comment:
I think Black has gone all out on the whitespace here because of the trailing commas

Reply:
you're right!